src/ServiceControl.Monitoring/Http/DiagramApiModule.cs

```diff
@@ -0,0 +1,65 @@
+namespace ServiceControl.Monitoring.Http
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Nancy;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+    using Timings;
+
+    /// <summary>
+    /// Exposes ServiceControl.Monitoring metrics needed for in endpoint overview page.
+    /// </summary>
+    public class DiagramsApiModule : NancyModule
+    {
+        /// <summary>
+        /// Initializes the metric API module.
+        /// </summary>
+        public DiagramsApiModule(ProcessingTimeStore processingTimeStore, CriticalTimeStore criticalTimeStore) : base("/diagrams")
+        {
+            After.AddItemToEndOfPipeline(ctx => ctx.Response
+                .WithHeader("Access-Control-Allow-Origin", "*")
+                .WithHeader("Access-Control-Allow-Methods", "POST,GET")
+                .WithHeader("Access-Control-Allow-Headers", "Accept, Origin, Content-type"));
+
+            Get["/data"] = x =>
+            {
+                var now = DateTime.UtcNow;
+
+                var processingTimes = processingTimeStore.GetTimings(now);
+                var criticalTimes = criticalTimeStore.GetTimings(now);
+
+                var endpointNames = processingTimes.Select(e => e.EndpointName)
+                    .Union(criticalTimes.Select(e => e.EndpointName));
+
+                var endpointsResult = new List<JObject>();
+
+                foreach (var endpointName in endpointNames)
+                {
+                    var processingTime = processingTimes.FirstOrDefault(pt => pt.EndpointName == endpointName);
```

Comment:
Would it make sense to have 3 different calls here? Get Endpoints, Get Processing Time, Get Critical Time? That will change the UI from 2 remote calls to 5 but it gives us greater flexibility.